kong/tools/database_cache.lua

```diff
@@ -120,21 +121,42 @@ function _M.all_apis_by_dict_key()
 end
 
 function _M.get_or_set(key, cb)
+  local lock = resty_lock:new("cache_locks", {
+    exptime = 10,
+    timeout = 10
```

Comment:
I think the default value of `5` for `timeout` is safer. If anything this value should probably be **decreased** down to `1` or even `0`, but not increased, especially since `get_or_set()` is called in the hotest code path of our application (in the resolver's `find_api`. `timeout` prevents a call to `lock` or `unlock` from hanging too long.

Secondly: the lock should not be created unless the case hit is a miss. Currently, this lock is created *before* even trying to access the cache (a few lines down).